tests/test_component.py

```diff
@@ -204,6 +204,60 @@ def test_component_with_relative_paths_as_subcomponent(
 
         self.assertIn('<input type="text" name="variable" value="hello">', rendered, rendered)
 
+    def test_component_inside_slot(self):
```

Comment:
While we do have components inside slot fills in some other tests, I didnt't find any test that'd test explicitly for that, so added this one.